src/common/libmissing/pipe2.c

```diff
@@ -15,14 +15,26 @@
 #include <fcntl.h>
 #include <errno.h>
 
+#include "src/common/libutil/fdutils.h"
+
 #include "pipe2.h"
 
 static int setflags (int fd, int flags)
 {
-    int oflags;
-    if ((oflags = fcntl (fd, F_GETFL)) < 0
-        || fcntl (fd, F_SETFL, oflags | flags) < 0)
+    int valid_flags = O_CLOEXEC | O_NONBLOCK;
+
+    if ((flags & ~valid_flags)) {
+        errno = EINVAL;
         return -1;
+    }
+    if ((flags & O_CLOEXEC)) {
+        if (fd_set_cloexec (fd) < 0)
+            return -1;
+    }
+    if ((flags & O_NONBLOCK)) {
+        if (fd_set_nonblocking (fd) < 0)
+            return -1;
+    }
```

Comment:
what you did here doesn't match your commit message.  Not sure if you changed mind on what to do?  hopefully not a rebase mistake?

Reply:
Those are just helper functions that do what it says in the commit message but I guess that could be made more clear.

Reply:
Just pushed a tweak to the commit message

Reply:
Ahh, you know what it is.  Seeing the very generic prior implementation, the appearance is that more/other flags would have been supported.

But I now see that `pipe2()` only supports three flags, one of which appears to be linux only.   so the new `setflags()` supports every flag for `pipe2()`